broadinstitute/wfl
# [GH-1566] Add workload info to logs

### Purpose


We would like for the logs that are printed in our Kubernetes pods
to include some information about the workloads they are about
to allow for better debugging and metrics for WFL.

- https://broadinstitute.atlassian.net/browse/GH-1566
- https://broadinstitute.atlassian.net/browse/GH-1571

### Changes


- Simplify logging macros and provide more context.
- Support adding extra fields to the log record payload.
- Log the workload UUID and workload labels to some log sites.
- Make logging of random Java objects a little more robust.
- Make most database access wrappers fnz instead of macroze.
- Name and invoke the log API more consistently.
- Hide some GCP/Stackdriver details.
- Help `wfl.service.clio/post` handle non-JSON responses better.
- Pass `throw` in `wfl.service.clio/add` a map instead of a seq.
- Log `wfl.module.aou` surprises as errors instead of warnings.
- Log `wfl.module.sg` surprises as errors instead of warnings.
- Do not discard the `wfl.server/update-workloads` future.
- Delete `wfl.integration.datarepo-test/replace-urls-with-file-ids`.
- Delete `wfl.system.v1-endpoint-test/ingest-illumina-genotyping-array-files`.
- Organize the `wfl.server` namespace to make bugs more evident.
- Format `clojure.core/ns` forms more consistently.
- Break down some long lines of code for readability.

### Review Instructions


- You might want to click `Hide whitespace` for this.
- Start reading in `wfl.jdbc`.
- Then glance through `wfl.log` itself.
- Then you could visit all of the `(log/` instances in the code.
- Run a local server and watch the log records.
- Run a workload and check the logs.
- See logs that have workload UUIDs and labels included.
- Run system tests on a local server with `"DEBUG"` enabled.

### Not done yet

- More of our `:labels` keys should be `::log/labels`.
- Simplify log calls by consing fewer strings.
- I think it would be nice to log expressions inline -- or in other words …
- Treat `(log/debug ...)` as an expression rather than a statement.
